rotkehlchen/externalapis/cowswap.py

```diff
@@ -0,0 +1,94 @@
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+ORDER_TYPE_CHANGE_TS: Final = Timestamp(1710806400)
+CHAIN_MAPPING: Final = {
+    SupportedBlockchain.ETHEREUM: 'mainnet',
+    SupportedBlockchain.GNOSIS: 'xdai',
+    SupportedBlockchain.ARBITRUM_ONE: 'arbitrum_one',
+}
+
+
+class CowswapAPI:
+    """This is the Cowswap API interface.
+
+    https://api.cow.fi/docs/
+    """
+
+    def __init__(self, database: 'DBHandler', chain: 'SupportedBlockchain') -> None:
+        self.database = database
+        self.session = requests.session()
+
+        if chain not in CHAIN_MAPPING:
+            log.warning(f'Encountered unsupported Cowswap API chain {chain}. Falling back to mainnet.')  # noqa: E501
+            chain = SupportedBlockchain.ETHEREUM
+
+        self.api_url = f'https://api.cow.fi/{CHAIN_MAPPING[chain]}/api/v1'
+
+    def _query(self, endpoint: str) -> dict[str, Any]:
+        """Query cowswap API endpoint."""
+        query_str = f'{self.api_url}/{endpoint}'
+        log.debug(f'Querying Cowswap API {query_str}')
+        timeout = CachedSettings().get_timeout_tuple()
+        try:
+            response = self.session.get(url=query_str, timeout=timeout)
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(f'Querying {query_str} failed due to {e!s}') from e
+
+        if response.status_code != 200:
+            raise RemoteError(f'Cowswap API request {response.url} failed with HTTP status code {response.status_code} and text {response.text}')  # noqa: E501
+
+        try:
+            json_ret = json.loads(response.text)
+        except json.decoder.JSONDecodeError as e:
+            raise RemoteError(f'Cowswap API returned invalid JSON response: {response.text}') from e  # noqa: E501
+
+        return json_ret
+
+    def query_order_data(self, order_uid: str) -> tuple[str, str]:
+        """Query offchain order data from the database or the api.
+        Returns raw_fee_amount and order_type as a tuple.
+        """
+        with self.database.conn.read_ctx() as cursor:
+            result = cursor.execute(
+                'SELECT order_type, raw_fee_amount FROM cowswap_orders WHERE identifier = ?',
+                (order_uid,),
+            ).fetchone()
+
+        if result is not None:
+            raw_fee_amount = result[0]
+            order_type = result[1]
+        else:
+            data = self._query(f'orders/{order_uid}')
+            try:
+                if iso8601ts_to_timestamp(data['creationDate']) < ORDER_TYPE_CHANGE_TS:
+                    raw_fee_amount = data['executedFeeAmount']
+                    order_type = data['class']
+                else:
+                    raw_fee_amount = data['executedSurplusFee']
+                    order_type = json.loads(data['fullAppData'])['metadata']['orderClass']['orderClass']  # noqa: E501
+            except (KeyError, json.decoder.JSONDecodeError) as e:
```

Comment:
you also need to handle the `ValueError` that doing `int(x)` will raise if `x` is not a stringified int
```suggestion
            except (KeyError, json.decoder.JSONDecodeError, ValueError) as e:
```